drivers/mfd/mfd_axp192.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2023 Martin Kiepfer <mrmarteng@teleschirm.org>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT x_powers_axp192
+
+#include <errno.h>
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(mfd_axp192, CONFIG_MFD_LOG_LEVEL);
+
+/* Chip ID value */
+#define AXP192_CHIP_ID (0x03u)
+
+/* Registers definitions */
+#define AXP192_REG_CHIP_ID (0x03u)
+
+struct mfd_axp192_config {
+	struct i2c_dt_spec i2c;
+};
+
+static int mfd_axp192_init(const struct device *dev)
+{
+	const struct mfd_axp192_config *config = dev->config;
+	uint8_t chip_id;
+	int ret;
+
+	LOG_DBG("Initializing instance");
+
+	if (!i2c_is_ready_dt(&config->i2c)) {
+		LOG_ERR("I2C bus not ready");
+		return -ENODEV;
+	}
+
+	/* Check if axp192 chip is available */
+	ret = i2c_reg_read_byte_dt(&config->i2c, AXP192_REG_CHIP_ID, &chip_id);
+	if (ret < 0) {
+		LOG_ERR("I2C read failed (%d)", ret);
+		return ret;
+	}
+
+	if (chip_id != AXP192_CHIP_ID) {
+		LOG_ERR("Invalid Chip detected (%d)", chip_id);
+		ret = -EINVAL;
+	}
+	LOG_INF("Chip detected");
+
+	return ret;
+}
```

Comment:
I don't think it's really useful to add a MFD device just to check if the chip is there. It's just a waste of ROM for nothing. MFD is meant to:

1. Provide a common access API (if needed, it's not always the case)
2. Initialize common parameters of the device

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/58809#discussion_r1221072034 -- I think the idea was that eventually this would allow to support GPIO, ADC, or fuel gauge, since the AXP192 has that too.

Reply:
So it should be added when required. DT structure should use the MFD layout, though.

Reply:
I was following an advice from another reviewer to add the MFD layer now. With the GPIO driver the MFD layer incudes much more functionality (https://github.com/MrMarteng/zephyr/tree/axp192_gpio). I am ready to create a separate PR for the GPIO driver, but I thought it is better to finish this regulator PR first.

>>DT structure should use the MFD layout, though.
Can you describe in more detail, what you mean with that? 
Should all DT specifications including the regulator, be part of the MFD yaml file?

Reply:
The MFD driver should be added only if needed, so it can be added later. Also note that you don't have to take MFD I2C dt-spec. See e.g. nPM6001. By DT layout I mean:

```
i2c@xxx {
    pmic@yy {
         compatible = "vnd,pmic";
         ...
         regulators {
               compatible = "vnd,pmic-regulator";
               ...
         };
};
```

Reply:
This should be fixed meanwhile